[Bug]: terraform import aws_opensearchserverless_collection crashes with SIGSEGV

### Terraform Core Version

1.5.7

### AWS Provider Version

5.18.1

### Affected Resource(s)

* aws_opensearchserverless_collection

### Expected Behavior

Resource is imported

### Actual Behavior

Crashes

### Relevant Error/Panic Output Snippet

```shell
$ terraform import aws_opensearchserverless_collection.xxx yyy
aws_opensearchserverless_collection.xxx: Importing from ID "yyy"...
aws_opensearchserverless_collection.xxx: Import prepared!
data.aws_caller_identity.current: Reading...
  Prepared aws_opensearchserverless_collection for import
aws_opensearchserverless_collection.xxx: Refreshing state... [id=yyy]
╷
│ Error: Request cancelled
│ 
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵

╷
│ Error: Request cancelled
│ 
│   with data.aws_caller_identity.current,
│   on /...redacted/opensearch.tf line 57, in data "aws_caller_identity" "current":
│   57: data "aws_caller_identity" "current" {}
│ 
│ The plugin.(*GRPCProvider).ReadDataSource request was cancelled.
╵


Stack trace from the terraform-provider-aws_v5.18.1_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa5deef8]

goroutine 353 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless.(*resourceCollection).Read(0xc0005c0001?, {0xee42680, 0xc0005fc2a0}, {{{{0xee4ec38, 0xc0005fc060}, {0xc120b40, 0xc0005bdf50}}, {0xee57f38, 0xc001794f50}}, 0xc0000120f0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless/collection.go:211 +0x1b8
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.(*wrappedResource).Read.func1({0xee42680?, 0xc0005fc2a0?}, {{{{0xee4ec38, 0xc0005fc060}, {0xc120b40, 0xc0005bdf50}}, {0xee57f38, 0xc001794f50}}, 0xc0000120f0, {{{0x0, ...}, ...}, ...}}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:229 +0x89
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.interceptedHandler[...].func1({{{{0xee4ec38, 0xc0005fc060}, {0xc120b40, 0xc0005bdf50}}, {0xee57f38, 0xc001794f50}}, 0xc0000120f0, {{{0x0, 0x0}, {0x0, ...}}, ...}}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:119 +0x2ff
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.(*wrappedResource).Read(0xc0005b0480, {0xee42680, 0xc0005bd050}, {{{{0xee4ec38, 0xc0005fc060}, {0xc120b40, 0xc0005bdf50}}, {0xee57f38, 0xc001794f50}}, 0xc0000120f0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:233 +0x157
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0023fd8c0, {0xee42680, 0xc0005bd050}, 0xc0005bd0e0, 0xc001c955a8)
	github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwserver/server_readresource.go:101 +0x677
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ReadResource(0xc0023fd8c0, {0xee42680?, 0xc0005bcf00?}, 0xc0005b0280)
	github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/proto5server/server_readresource.go:56 +0x41a
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0xee425d8?, {0xee42680?, 0xc0005bcbd0?}, 0xc0005b0280)
	github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf5muxserver/mux_server_ReadResource.go:35 +0x1b5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc003faa8c0, {0xee42680?, 0xc0005bc1e0?}, 0xc0005b21e0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:789 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xd82c9a0?, 0xc003faa8c0}, {0xee42680, 0xc0005bc1e0}, 0xc00051c000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:431 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00029c780, {0xee4fb80, 0xc0041d01a0}, 0xc0005c8000, 0xc002505020, 0x15205648, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1360 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc00029c780, {0xee4fb80, 0xc0041d01a0}, 0xc0005c8000, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1737 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.57.0/server.go:982 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.57.0/server.go:980 +0x18c

Error: The terraform-provider-aws_v5.18.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

# empty, waiting for the import
resource "aws_opensearchserverless_collection" "xxx" {
}

### Steps to Reproduce

1. Create an AOSS collection.
2. `$ terraform import aws_opensearchserverless_collection.xxx yyy`

### Debug Output

```
2023-09-27T15:46:04.676-0700 [INFO]  Terraform version: 1.5.7
2023-09-27T15:46:04.676-0700 [DEBUG] using github.com/hashicorp/go-tfe v1.26.0
2023-09-27T15:46:04.676-0700 [DEBUG] using github.com/hashicorp/hcl/v2 v2.16.2
2023-09-27T15:46:04.676-0700 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.0
2023-09-27T15:46:04.676-0700 [DEBUG] using github.com/zclconf/go-cty v1.12.2
2023-09-27T15:46:04.676-0700 [INFO]  Go runtime version: go1.20.7
2023-09-27T15:46:04.676-0700 [INFO]  CLI args: []string{"terraform", "import", "aws_opensearchserverless_collection.xxx", "yyy"}
2023-09-27T15:46:04.676-0700 [DEBUG] Attempting to open CLI config file: (redacted)
2023-09-27T15:46:04.676-0700 [INFO]  Loading CLI configuration from (redacted)
2023-09-27T15:46:04.678-0700 [INFO]  CLI command args: []string{"import", "aws_opensearchserverless_collection.xxx", "yyy"}
2023-09-27T15:46:04.681-0700 [DEBUG] New state was assigned lineage "9b87fbf0-d42a-bd95-3f0d-022b4ef72319"
2023-09-27T15:46:06.092-0700 [DEBUG] checking for provisioner in "."
2023-09-27T15:46:06.097-0700 [DEBUG] checking for provisioner in "/usr/bin"
2023-09-27T15:46:06.099-0700 [DEBUG] Building and walking import graph
2023-09-27T15:46:06.100-0700 [DEBUG] ProviderTransformer: "aws_opensearchserverless_security_policy.xxx (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2023-09-27T15:46:06.100-0700 [DEBUG] ProviderTransformer: "aws_opensearchserverless_security_policy.example (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2023-09-27T15:46:06.100-0700 [DEBUG] ProviderTransformer: "aws_opensearchserverless_access_policy.xxx (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2023-09-27T15:46:06.100-0700 [DEBUG] ProviderTransformer: "data.aws_caller_identity.current (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2023-09-27T15:46:06.100-0700 [DEBUG] ProviderTransformer: "aws_opensearchserverless_collection.xxx (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2023-09-27T15:46:06.100-0700 [DEBUG] created provider logger: level=debug
2023-09-27T15:46:06.100-0700 [INFO]  provider: configuring client automatic mTLS
2023-09-27T15:46:06.120-0700 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5]
2023-09-27T15:46:06.121-0700 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5 pid=689416
2023-09-27T15:46:06.121-0700 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5
2023-09-27T15:46:06.250-0700 [INFO]  provider.terraform-provider-aws_v5.18.1_x5: configuring server automatic mTLS: timestamp=2023-09-27T15:46:06.249-0700
2023-09-27T15:46:06.274-0700 [DEBUG] provider: using plugin: version=5
2023-09-27T15:46:06.274-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: plugin address: network=unix address=/tmp/plugin2294982161 timestamp=2023-09-27T15:46:06.273-0700
2023-09-27T15:46:07.158-0700 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-09-27T15:46:07.167-0700 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5 pid=689416
2023-09-27T15:46:07.167-0700 [DEBUG] provider: plugin exited
2023-09-27T15:46:07.167-0700 [DEBUG] ReferenceTransformer: "aws_opensearchserverless_security_policy.example (expand)" references: []
2023-09-27T15:46:07.167-0700 [DEBUG] ReferenceTransformer: "aws_opensearchserverless_access_policy.xxx (expand)" references: [data.aws_caller_identity.current (expand)]
2023-09-27T15:46:07.167-0700 [DEBUG] ReferenceTransformer: "data.aws_caller_identity.current (expand)" references: []
2023-09-27T15:46:07.167-0700 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/aws\"]" references: []
2023-09-27T15:46:07.167-0700 [DEBUG] ReferenceTransformer: "aws_opensearchserverless_collection.xxx (expand)" references: []
2023-09-27T15:46:07.167-0700 [DEBUG] ReferenceTransformer: "aws_opensearchserverless_security_policy.xxx (expand)" references: []
2023-09-27T15:46:07.168-0700 [DEBUG] Starting graph walk: walkImport
2023-09-27T15:46:07.168-0700 [DEBUG] created provider logger: level=debug
2023-09-27T15:46:07.168-0700 [INFO]  provider: configuring client automatic mTLS
2023-09-27T15:46:07.176-0700 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5]
2023-09-27T15:46:07.176-0700 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5 pid=689450
2023-09-27T15:46:07.176-0700 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5
2023-09-27T15:46:07.305-0700 [INFO]  provider.terraform-provider-aws_v5.18.1_x5: configuring server automatic mTLS: timestamp=2023-09-27T15:46:07.304-0700
2023-09-27T15:46:07.328-0700 [DEBUG] provider: using plugin: version=5
2023-09-27T15:46:07.328-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: plugin address: address=/tmp/plugin3336947544 network=unix timestamp=2023-09-27T15:46:07.328-0700
2023-09-27T15:46:08.089-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Configuring Terraform AWS Provider: tf_mux_provider=*schema.GRPCProviderServer tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 @caller=github.com/hashicorp/terraform-provider-aws/internal/conns/config.go:119 @module=aws tf_rpc=ConfigureProvider timestamp=2023-09-27T15:46:08.089-0700
2023-09-27T15:46:08.089-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Resolving credentials provider: @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:47 tf_mux_provider=*schema.GRPCProviderServer tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ConfigureProvider @module=aws.aws-base tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 timestamp=2023-09-27T15:46:08.089-0700
2023-09-27T15:46:08.089-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Loading configuration: tf_provider_addr=registry.terraform.io/hashicorp/aws @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:47 @module=aws.aws-base tf_mux_provider=*schema.GRPCProviderServer tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 tf_rpc=ConfigureProvider timestamp=2023-09-27T15:46:08.089-0700
2023-09-27T15:46:08.089-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Retrieving credentials: tf_mux_provider=*schema.GRPCProviderServer tf_provider_addr=registry.terraform.io/hashicorp/aws @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:47 @module=aws.aws-base tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 tf_rpc=ConfigureProvider timestamp=2023-09-27T15:46:08.089-0700
2023-09-27T15:46:08.089-0700 [INFO]  provider.terraform-provider-aws_v5.18.1_x5: Retrieved credentials: @module=aws.aws-base tf_aws.credentials_source=EnvConfigCredentials tf_mux_provider=*schema.GRPCProviderServer tf_provider_addr=registry.terraform.io/hashicorp/aws @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:39 tf_rpc=ConfigureProvider tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 timestamp=2023-09-27T15:46:08.089-0700
2023-09-27T15:46:08.089-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Loading configuration: tf_mux_provider=*schema.GRPCProviderServer @module=aws.aws-base tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 tf_rpc=ConfigureProvider @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:47 timestamp=2023-09-27T15:46:08.089-0700
2023-09-27T15:46:08.089-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Retrieving caller identity from STS: tf_mux_provider=*schema.GRPCProviderServer tf_provider_addr=registry.terraform.io/hashicorp/aws @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:47 @module=aws.aws-base tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 tf_rpc=ConfigureProvider timestamp=2023-09-27T15:46:08.089-0700
2023-09-27T15:46:08.090-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: HTTP Request Sent: @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:47 @module=aws.aws-base http.request.header.x_amz_date=20230927T224608Z http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.5.7 (+https://www.terraform.io) terraform-provider-aws/5.18.1 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go-v2/1.21.0 os/linux lang/go#1.20.7 md/GOOS#linux md/GOARCH#amd64 api/sts#1.22.0" tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ConfigureProvider aws.region=us-west-2 aws.service=STS http.request.body="Action=GetCallerIdentity&Version=2011-06-15
" http.request.header.amz_sdk_request="attempt=1; max=25" tf_mux_provider=*schema.GRPCProviderServer tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 aws.operation=GetCallerIdentity http.method=POST http.request.header.content_type=application/x-www-form-urlencoded http.request.header.x_amz_security_token=***** http.url=https://sts.us-west-2.amazonaws.com/ net.peer.name=sts.us-west-2.amazonaws.com aws.sdk=aws-sdk-go-v2 http.request.header.amz_sdk_invocation_id=7e64ae54-a241-451e-81ec-a438126f53d3 http.request.header.authorization="redacted" http.request_content_length=43 timestamp=2023-09-27T15:46:08.090-0700
2023-09-27T15:46:08.248-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: HTTP Response Received: tf_provider_addr=registry.terraform.io/hashicorp/aws aws.region=us-west-2 aws.service=STS http.response.body="<GetCallerIdentityResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
" http.response_content_length=484 http.status_code=200 tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 http.response.header.content_type=text/xml tf_rpc=ConfigureProvider @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:47 aws.operation=GetCallerIdentity aws.sdk=aws-sdk-go-v2 http.response.header.x_amzn_requestid=8b8f5a28-1056-4381-846b-37a7fb23ad79 tf_mux_provider=*schema.GRPCProviderServer @module=aws.aws-base http.duration=156 http.response.header.date="Wed, 27 Sep 2023 22:46:07 GMT" timestamp=2023-09-27T15:46:08.247-0700
2023-09-27T15:46:08.248-0700 [INFO]  provider.terraform-provider-aws_v5.18.1_x5: Retrieved caller identity from STS: @module=aws.aws-base tf_mux_provider=*schema.GRPCProviderServer tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ConfigureProvider @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:39 tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 timestamp=2023-09-27T15:46:08.248-0700
2023-09-27T15:46:08.248-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Creating AWS SDK v1 session: @module=aws tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ConfigureProvider tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 @caller=github.com/hashicorp/terraform-provider-aws/internal/conns/config.go:141 tf_mux_provider=*schema.GRPCProviderServer timestamp=2023-09-27T15:46:08.248-0700
2023-09-27T15:46:08.248-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Retrieving AWS account details: tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ConfigureProvider @caller=github.com/hashicorp/terraform-provider-aws/internal/conns/config.go:156 @module=aws tf_mux_provider=*schema.GRPCProviderServer timestamp=2023-09-27T15:46:08.248-0700
2023-09-27T15:46:08.248-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: Retrieving caller identity from STS: tf_rpc=ConfigureProvider @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:47 tf_provider_addr=registry.terraform.io/hashicorp/aws @module=aws.aws-base tf_mux_provider=*schema.GRPCProviderServer tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 timestamp=2023-09-27T15:46:08.248-0700
(redacted)
2023-09-27T15:46:08.281-0700 [INFO]  provider.terraform-provider-aws_v5.18.1_x5: Retrieved caller identity from STS: @caller=github.com/hashicorp/aws-sdk-go-base/v2@v2.0.0-beta.35/logging/tf_logger.go:39 @module=aws.aws-base tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ConfigureProvider tf_mux_provider=*schema.GRPCProviderServer tf_req_id=bedb42b1-dae4-01a7-b7fd-3b193fd1c199 timestamp=2023-09-27T15:46:08.281-0700
2023-09-27T15:46:08.283-0700 [DEBUG] ReferenceTransformer: "aws_opensearchserverless_security_policy.example" references: []
2023-09-27T15:46:08.283-0700 [DEBUG] ReferenceTransformer: "data.aws_caller_identity.current" references: []
2023-09-27T15:46:08.284-0700 [DEBUG] ReferenceTransformer: "aws_opensearchserverless_security_policy.xxx" references: []
2023-09-27T15:46:08.284-0700 [DEBUG] ReferenceTransformer: "aws_opensearchserverless_collection.xxx (import id \"yyy\")" references: []
aws_opensearchserverless_collection.xxx: Importing from ID "yyy"...
data.aws_caller_identity.current: Reading...
aws_opensearchserverless_collection.xxx: Import prepared!
  Prepared aws_opensearchserverless_collection for import
aws_opensearchserverless_collection.xxx: Refreshing state... [id=yyy]
2023-09-27T15:46:08.287-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: HTTP Request Sent: @module=aws aws.sdk=aws-sdk-go aws.signing_region=us-east-1 http.request.header.content_type="application/x-www-form-urlencoded; charset=utf-8" http.request_content_length=43 @caller=github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2@v2.0.0-beta.36/logger.go:96 http.method=POST http.request.body="Action=GetCallerIdentity&Version=2011-06-15
" http.request.header.x_amz_security_token=***** net.peer.name=sts.amazonaws.com aws.operation=GetCallerIdentity aws.region=us-west-2 http.request.header.authorization="redacted" http.url=https://sts.amazonaws.com/ tf_mux_provider=*proto5server.Server aws.service=STS http.flavor=1.1 http.request.header.x_amz_date=20230927T224608Z http.user_agent="APN/1.0 HashiCorp/1.0 Terraform/1.5.7 (+https://www.terraform.io) terraform-provider-aws/5.18.1 (+https://registry.terraform.io/providers/hashicorp/aws) aws-sdk-go/1.45.14 (go1.20.7; linux; amd64)" tf_data_source_type=aws_caller_identity tf_provider_addr=registry.terraform.io/hashicorp/aws tf_req_id=f958499b-544c-584e-45ba-7d2949bc7975 tf_rpc=ReadDataSource timestamp=2023-09-27T15:46:08.287-0700
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: panic: runtime error: invalid memory address or nil pointer dereference
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa5deef8]
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: goroutine 353 [running]:
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless.(*resourceCollection).Read(0xc002120001?, {0xee42680, 0xc002156570}, {{{{0xee4ec38, 0xc002156330}, {0xc120b40, 0xc002156240}}, {0xee57f38, 0xc003a8e6e0}}, 0xc0001b4098, ...}, ...)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless/collection.go:211 +0x1b8
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.(*wrappedResource).Read.func1({0xee42680?, 0xc002156570?}, {{{{0xee4ec38, 0xc002156330}, {0xc120b40, 0xc002156240}}, {0xee57f38, 0xc003a8e6e0}}, 0xc0001b4098, {{{0x0, ...}, ...}, ...}}, ...)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:229 +0x89
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.interceptedHandler[...].func1({{{{0xee4ec38, 0xc002156330}, {0xc120b40, 0xc002156240}}, {0xee57f38, 0xc003a8e6e0}}, 0xc0001b4098, {{{0x0, 0x0}, {0x0, ...}}, ...}}, ...)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:119 +0x2ff
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.(*wrappedResource).Read(0xc0021044c0, {0xee42680, 0xc00067d320}, {{{{0xee4ec38, 0xc002156330}, {0xc120b40, 0xc002156240}}, {0xee57f38, 0xc003a8e6e0}}, 0xc0001b4098, ...}, ...)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:233 +0x157
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0023998c0, {0xee42680, 0xc00067d320}, 0xc00067d3b0, 0xc0021775a8)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwserver/server_readresource.go:101 +0x677
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ReadResource(0xc0023998c0, {0xee42680?, 0xc00067d1d0?}, 0xc0021042c0)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/proto5server/server_readresource.go:56 +0x41a
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0xee425d8?, {0xee42680?, 0xc00067cea0?}, 0xc0021042c0)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf5muxserver/mux_server_ReadResource.go:35 +0x1b5
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0008892c0, {0xee42680?, 0xc00067c4e0?}, 0xc0006741e0)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:789 +0x4b1
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xd82c9a0?, 0xc0008892c0}, {0xee42680, 0xc00067c4e0}, 0xc000a88070, 0x0)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:431 +0x170
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00002ab40, {0xee4fb80, 0xc0023bdba0}, 0xc00067afc0, 0xc0024f3a70, 0x15205648, 0x0)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	google.golang.org/grpc@v1.57.0/server.go:1360 +0xe23
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: google.golang.org/grpc.(*Server).handleStream(0xc00002ab40, {0xee4fb80, 0xc0023bdba0}, 0xc00067afc0, 0x0)
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	google.golang.org/grpc@v1.57.0/server.go:1737 +0xa36
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: google.golang.org/grpc.(*Server).serveStreams.func1.1()
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	google.golang.org/grpc@v1.57.0/server.go:982 +0x98
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2023-09-27T15:46:08.440-0700 [DEBUG] provider.terraform-provider-aws_v5.18.1_x5: 	google.golang.org/grpc@v1.57.0/server.go:980 +0x18c
2023-09-27T15:46:08.451-0700 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.18.1/linux_amd64/terraform-provider-aws_v5.18.1_x5 pid=689450 error="exit status 2"
2023-09-27T15:46:08.451-0700 [ERROR] plugin.(*GRPCProvider).ReadResource: error="rpc error: code = Canceled desc = context canceled"
2023-09-27T15:46:08.451-0700 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Canceled desc = context canceled"
2023-09-27T15:46:08.451-0700 [ERROR] vertex "import aws_opensearchserverless_collection.xxx result" error: Request cancelled
2023-09-27T15:46:08.451-0700 [ERROR] plugin.(*GRPCProvider).ReadDataSource: error="rpc error: code = Canceled desc = context canceled"
2023-09-27T15:46:08.451-0700 [ERROR] vertex "aws_opensearchserverless_collection.xxx (import id \"yyy\")" error: Request cancelled
2023-09-27T15:46:08.451-0700 [ERROR] vertex "data.aws_caller_identity.current" error: Request cancelled
2023-09-27T15:46:08.451-0700 [ERROR] vertex "aws_opensearchserverless_collection.xxx (expand)" error: Request cancelled
2023-09-27T15:46:08.451-0700 [ERROR] vertex "data.aws_caller_identity.current (expand)" error: Request cancelled
```

### Panic Output

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa5deef8]

goroutine 353 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless.(*resourceCollection).Read(0xc0005c0001?, {0xee42680, 0xc0005fc2a0}, {{{{0xee4ec38, 0xc0005fc060}, {0xc120b40, 0xc0005bdf50}}, {0xee57f38, 0xc001794f50}}, 0xc0000120f0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless/collection.go:211 +0x1b8
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.(*wrappedResource).Read.func1({0xee42680?, 0xc0005fc2a0?}, {{{{0xee4ec38, 0xc0005fc060}, {0xc120b40, 0xc0005bdf50}}, {0xee57f38, 0xc001794f50}}, 0xc0000120f0, {{{0x0, ...}, ...}, ...}}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:229 +0x89
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.interceptedHandler[...].func1({{{{0xee4ec38, 0xc0005fc060}, {0xc120b40, 0xc0005bdf50}}, {0xee57f38, 0xc001794f50}}, 0xc0000120f0, {{{0x0, 0x0}, {0x0, ...}}, ...}}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:119 +0x2ff
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.(*wrappedResource).Read(0xc0005b0480, {0xee42680, 0xc0005bd050}, {{{{0xee4ec38, 0xc0005fc060}, {0xc120b40, 0xc0005bdf50}}, {0xee57f38, 0xc001794f50}}, 0xc0000120f0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:233 +0x157
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0023fd8c0, {0xee42680, 0xc0005bd050}, 0xc0005bd0e0, 0xc001c955a8)
	github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/fwserver/server_readresource.go:101 +0x677
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ReadResource(0xc0023fd8c0, {0xee42680?, 0xc0005bcf00?}, 0xc0005b0280)
	github.com/hashicorp/terraform-plugin-framework@v1.4.0/internal/proto5server/server_readresource.go:56 +0x41a
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0xee425d8?, {0xee42680?, 0xc0005bcbd0?}, 0xc0005b0280)
	github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf5muxserver/mux_server_ReadResource.go:35 +0x1b5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc003faa8c0, {0xee42680?, 0xc0005bc1e0?}, 0xc0005b21e0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:789 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xd82c9a0?, 0xc003faa8c0}, {0xee42680, 0xc0005bc1e0}, 0xc00051c000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:431 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00029c780, {0xee4fb80, 0xc0041d01a0}, 0xc0005c8000, 0xc002505020, 0x15205648, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1360 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc00029c780, {0xee4fb80, 0xc0041d01a0}, 0xc0005c8000, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1737 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.57.0/server.go:982 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.57.0/server.go:980 +0x18c
```

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

--- a/.changelog/33918.txt
+++ b/.changelog/33918.txt
@@ -1,3 +1,3 @@
 ```release-note:bug
-resource/ws_opensearchserverless_collection: Fix crash when error is returned
+resource/aws_opensearchserverless_collection: Fix crash when error is returned
 ```

--- a/.changelog/33918.txt
+++ b/.changelog/33918.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/ws_opensearchserverless_collection: Fix crash when error is returned
+```

--- a/internal/service/opensearchserverless/collection.go
+++ b/internal/service/opensearchserverless/collection.go
@@ -208,6 +208,14 @@ func (r *resourceCollection) Read(ctx context.Context, req resource.ReadRequest,
 		return
 	}
 
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.OpenSearchServerless, create.ErrActionReading, ResNameCollection, state.ID.ValueString(), nil),
+			err.Error(),
+		)
+		return
+	}
+
 	state.ARN = flex.StringToFramework(ctx, out.Arn)
 	state.CollectionEndpoint = flex.StringToFramework(ctx, out.CollectionEndpoint)
 	state.DashboardEndpoint = flex.StringToFramework(ctx, out.DashboardEndpoint)

